datalad/distribution/create_sibling.py

```diff
@@ -146,46 +148,86 @@ class CreateSibling(Interface):
+                 recursion_limit=None,
                  existing='error', shared=False, ui=False,
                  as_common_datasrc=None,
                  publish_by_default=None,
                  publish_depends=None):
 
-        if sshurl is None:
-            raise ValueError("""insufficient information for target creation
-            (needs at least a dataset and a SSH URL).""")
+        # there is no point in doing anything further
+        not_supported_on_windows(
+            "Support for SSH connections is not yet implemented in Windows")
 
-        if name is None and \
-                (target_url is not None or
-                 target_pushurl is not None):
-            raise ValueError("""insufficient information for adding the target
-            as a sibling (needs at least a name)""")
+        if not sshurl:
+            raise InsufficientArgumentsError(
+                "need at least an SSH URL")
 
-        # shortcut
-        ds = require_dataset(dataset, check_installed=True,
-                             purpose='creating a sibling')
-
-        assert(ds is not None and sshurl is not None and ds.repo is not None)
-
-        # determine target parameters:
+        # check the login URL
         sshri = RI(sshurl)
-
         if not isinstance(sshri, SSHRI) \
                 and not (isinstance(sshri, URL) and sshri.scheme == 'ssh'):
             raise ValueError(
-                "Unsupported SSH URL: '{0}', use ssh://host/path or host:path syntax".format(
-                    sshurl))
+                "Unsupported SSH URL: '{0}', "
+                "use ssh://host/path or host:path syntax".format(sshurl))
 
         if not name:
-            # use the hostname
+            # use the hostname as default remote name
             name = sshri.hostname
             lgr.debug(
                 "No sibling name given, use URL hostname '%s' as sibling name",
                 name)
 
-        # TODO add check if such a remote already exists and act based on
-        # --existing
+        ds = require_dataset(dataset, check_installed=True,
+                             purpose='creating a sibling')
+
+        # use common sorting implementation to discover all subdatasets
+        content_by_ds, unavailable_paths = Interface._prep(
+            # the base data set is the only path
+            dataset=ds,
+            recursive=recursive,
+            recursion_limit=recursion_limit)
+        # dataset arg was tested before, only existing dataset should be reported
+        assert(not unavailable_paths)
+
+        # anal verification
+        assert(ds is not None and sshurl is not None and ds.repo is not None)
+
+        # TODO apply a future --since
+
+        # dataset instances
+        datasets = {p: Dataset(p) for p in content_by_ds}
+
+        # find datasets with existing remotes with the target name
+        remote_existing = [p for p in datasets
+                           if name in datasets[p].repo.get_remotes()]
```

Comment:
I don't get this. If we have an AnnexRepo instance, this will be called anyways. If we don't have one, we cannot have special remotes. What am I missing?